### PR TITLE
add support for TTCN-3 Language Extensions: Advanced Parameterization

### PR DESCRIPTION
**Pull request by akrzyz**
add support for TTCN-3 Language Extensions: Advanced Parameterization
extension is described in: ETSI ES 202 784 V1.6.1

new Units are added:
 Units/parser-ttcn.r/ttcn-template-type.d
 Units/parser-ttcn.r/ttcn-template-template.d
 Units/parser-ttcn.r/ttcn-template-function.d
 Units/parser-ttcn.r/ttcn-template-variables-and-constants.d

### CODE REVIEW

File: parsers/ttcn.c [hunk 1]
```diff
@@ -894,7 +897,7 @@ static int parseTypeDefBody (void)
 			/* SubTypeDef */
 			ungetToken();
 			/* Stop after type name, no need to parse ArrayDef, StringLength, etc */
-			return (parseType() && parseID(K_TYPE));
+			return parseType() && parseID(K_TYPE);
```
Comment (masatake) [inline parsers/ttcn.c]: Do we need this change for supporting advanced parameterization?
At least this chnage should go to a separated commit.

File: parsers/ttcn.c [hunk 2]
```diff
@@ -933,7 +936,9 @@ static void parseTTCN (void)
 			case T_TEMPLATE:
 				/* A.1.6.1.3 TemplateDef ::= "template" (Type | Signature) ID ... */
 				if (parseType() || parseSignature())
+                {
```
Comment (masatake) [inline parsers/ttcn.c]: Do we need this change for supporting advanced parameterization?
At least this chnage should go to a separated commit.

File: parsers/ttcn.c [hunk 3]
```diff
@@ -933,7 +936,9 @@ static void parseTTCN (void)
...
 					parseID(K_TEMPLATE);
+                }
```
Comment (masatake) [inline parsers/ttcn.c]: Do we need this change for supporting advanced parameterization?
At least this chnage should go to a separated commit.

File: parsers/ttcn.c (hunk 1 above)
Comment (akrzyz) [reply parsers/ttcn.c]: we don't need this, i removed this changed and squashed commits into one.

**Comment by masatake** [comment]
Thank you!

**Pull request merged**